Document the k6/experimental/tracing module (#1020)
grafana/k6-docs: src/data/markdown/docs/02 javascript api/07 k6-experimental/04 tracing.md

| [Client](/javascript-api/k6-experimental/tracing/client)                 | configurable Client that exposes instrumented HTTP operations and allows selectively instrumenting requests with tracing. |
| [Options](/javascript-api/k6-experimental/tracing/options)               | Options allows to configure the tracing instrumentation behavior.                                                         |

## Example

This example demonstrates how to use the tracing API to instrument every HTTP request made in a script with tracing information.

<CodeGroup labels={["example-tracing-module.js"]} lineNumbers={[]} showCopyButton={[true]}>

```javascript
import http from "k6/http";
import { check } from "k6";
import tracing from "k6/experimental/tracing";

// instrumentHTTP will ensure that all requests made by the http module
// from this point forward will have a trace context attached.
//
// The first argument is a configuration object that
// can be used to configure the tracer.
//
// Currently supported HTTP methods are: get, post, put, patch, head,
// del, options, and request.
tracing.instrumentHTTP({

> propagator: "w3c",

Reviewer: ```suggestion
  propagator: "w3c",
```

Also, maybe add a `// possible values: "w3c", "jaeger"`? This is mentioned and explained on other pages, but since this is the first example users will see, just to give a heads-up for what's possible.

});

export default () => {
 // the instrumentHTTP call in the init context replaced
 // the http module with a version that will automatically
 // attach a trace context to every request.
 //
 // Because the instrumentHTTP call was configured with the
 // w3c trace propagator, this request will as a result have
 // a `traceparent` header attached. 
 const res = http.get("http://httpbin.org/get", {
 headers: {

>  "X-Example-Header": "instrumented/get",

Reviewer: ```suggestion
   "X-Example-Header": "instrumented/get",
```

The below `"status is 200"` line should also be further indented. Though indentation is off in general in all examples here.

---
title: "tracing"
excerpt: "k6 Tracing experimental API"
---

<ExperimentalBlockquote />

> This experimental module instruments HTTP requests in k6 scripts with tracing information.

Reviewer 2: ```suggestion
With this experimental module, you can _instrument_ you HTTP requests so that they emit traces as the test runs.
```

Reviewer 2: It was quite a sage for me to understand what instrumentation means, and my first impression was that we should define it, but now I think we can settle to try to implicitly define it in the first sentence.

Does it look alright?

Author: This is perfect 👍🏻

| [instrumentHTTP](/javascript-api/k6-experimental/tracing/instrumenthttp) | instruments the k6 http module with tracing capabilities.                                                                 |
| [Client](/javascript-api/k6-experimental/tracing/client)                 | configurable Client that exposes instrumented HTTP operations and allows selectively instrumenting requests with tracing. |
| [Options](/javascript-api/k6-experimental/tracing/options)               | Options to configure the tracing instrumentation behavior.                                                         |

## Example

This example demonstrates how to use the tracing API to instrument every HTTP request made in a script with tracing information.

<CodeGroup labels={["example-tracing-module.js"]} lineNumbers={[]} showCopyButton={[true]}>

```javascript
import { check } from "k6";
import tracing from "k6/experimental/tracing";
import http from "k6/http";

// instrumentHTTP will ensure that all requests made by the http module
// from this point forward will have a trace context attached.
//
// The first argument is a configuration object that
// can be used to configure the tracer.
//

> // Currently supported HTTP methods are: get, post, put, patch, head,

Reviewer 3: nit: asyncRequest is not mentioned. I am not certain if we should have all of them listed as well :shrug:

Author: I forgot 🤦🏻 Also think you're right, we could explicitly mention the ones who aren't instead (only `batch`) 👍🏻

Reviewer 3: This seems to not have been fixed :thinking:

Reviewer 3: It was fixed but trhre is a second copy see https://github.com/grafana/k6-docs/pull/1020/files#r1102867581

Author: Indeed, I've forgotten it 👍🏻

> This example demonstrates how to use the tracing API to instrument every HTTP request made in a script with tracing information.

Reviewer 3: This example seems to be the same as the on in instrumentHTTP but with slightly differnt comments.

Can we maybe not have one here? Maybe link to the other if you desire (or from it to here)

Author: I'd prefer both, as each page could be accessed independently from the other. Still, I don't have a strong opinion on this, if you prefer it that way, please feel free to go ahead with the change 🙇🏻

---
title: "tracing"
excerpt: "k6 Tracing experimental API"
---

<ExperimentalBlockquote />

> With this experimental module, you can _instrument_ you HTTP requests so that they emit traces as the test runs.

Reviewer 2: ```suggestion
With this experimental module, you can _instrument_ your HTTP requests so that they emit traces as the test runs.
```

## API

| Class/Function                                                           | Description                                                                                                               |
| :----------------------------------------------------------------------- | :------------------------------------------------------------------------------------------------------------------------ |
| [instrumentHTTP](/javascript-api/k6-experimental/tracing/instrumenthttp) | instruments the k6 http module with tracing capabilities.                                                                 |
| [Client](/javascript-api/k6-experimental/tracing/client)                 | configurable Client that exposes instrumented HTTP operations and allows selectively instrumenting requests with tracing. |

> | [Options](/javascript-api/k6-experimental/tracing/options)               | Options to configure the tracing instrumentation behavior.                                                         |

Reviewer 3: I can't see this being an export.

Are you documenting it here just to have a reference? I think it is enough to link it from both `Client` and `instrumentHttp`

Author: Indeed, you're right this page doesn't need to be an entrypoint 👍

## Example

This example demonstrates how to use the tracing API to instrument every HTTP request made in a script with tracing information.

<CodeGroup labels={["example-tracing-module.js"]} lineNumbers={[]} showCopyButton={[true]}>

```javascript
import { check } from "k6";
import tracing from "k6/experimental/tracing";
import http from "k6/http";

// instrumentHTTP will ensure that all requests made by the http module
// from this point forward will have a trace context attached.
//
// The first argument is a configuration object that
// can be used to configure the tracer.
tracing.instrumentHTTP({
  // possible values: "w3c", "jaeger"
  propagator: "w3c",
});

export default () => {
  // the instrumentHTTP call in the init context replaced
  // the http module with a version that will automatically
  // attach a trace context to every request.
  //
  // Because the instrumentHTTP call was configured with the
  // w3c trace propagator, this request will as a result have
  // a `traceparent` header attached.
  const res = http.get("http://httpbin.org/get", {
    headers: {
      "X-Example-Header": "instrumented/get",
    },
  });

>   check(res, { "status is 200": (r) => r.status === 200 });

Reviewer 3: Do we really need to use `check` in every example?

Also I am kind of against status checks given that [`expectedStatus`](https://k6.io/docs/javascript-api/k6-http/expectedstatuses/) does the same thing and the default already checks for 200-399 status checks

Author: We don't need to, removing them 👍

};

> 

Reviewer 3: ```suggestion
```
